Windows nightlies builds again

Linking error:
> echo linking release/qtox.exe && i686-w64-mingw32-g++ -Wl,--dynamicbase -Wl,--nxcompat -Wl,-s -Wl,-subsystem,windows -o release/qtox.exe object_script.qtox.Release  -lglu32 -luser32 -lmingw32 -L/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib /home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libqtmain.a -lshell32 -L/home/jenkins/workspace/qTox_build_windows_x86_release/qtox/libs/lib -ltoxav -ltoxcore -ltoxencryptsave -lsodium -lvpx -lpthread -lavdevice -lavformat -lavcodec -lavutil -lswscale -lOpenAL32 -lopus -lqrencode -lsqlcipher -lcrypto -lopengl32 -lole32 -loleaut32 -lvfw32 -lws2_32 -liphlpapi -lgdi32 -lshlwapi -luuid -lstrmiids -ltoxgroup -ltoxmessenger -ltoxfriends -ltoxnetcrypto -ltoxdht -ltoxnetwork -ltoxcrypto /home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libQt5OpenGL.a /home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libQt5Svg.a /home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libQt5Widgets.a /home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libQt5Gui.a /home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libQt5Network.a /home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libQt5Xml.a /home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libQt5Sql.a /home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libQt5Core.a release/qtox_res.o 
linking release/qtox.exe
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x21): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0xda): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x231): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x3e5): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x529): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x7ef): more undefined references to `_imp__send@16' follow
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0xa40): undefined reference to `_imp__ioctlsocket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0xa64): undefined reference to `_imp__recv@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0xab4): undefined reference to `_imp__ioctlsocket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0xae3): undefined reference to `_imp__recv@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0xe16): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x1544): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x166e): undefined reference to `_imp__ioctlsocket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x1698): undefined reference to `_imp__recv@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x1707): undefined reference to `_imp__ioctlsocket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x173c): undefined reference to `_imp__recv@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x1c6a): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x200a): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x2584): undefined reference to `_imp__ioctlsocket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x25c9): undefined reference to `_imp__ioctlsocket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x25f0): undefined reference to `_imp__recv@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x268d): undefined reference to `_imp__ioctlsocket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x26b3): undefined reference to `_imp__recv@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x276d): undefined reference to `_imp__ioctlsocket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x2791): undefined reference to `_imp__recv@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x27cf): undefined reference to `_imp__ioctlsocket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x27f8): undefined reference to `_imp__recv@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x29c0): undefined reference to `_imp__socket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x2b39): undefined reference to `_imp__bind@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x2b55): undefined reference to `_imp__listen@8'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x2c50): undefined reference to `_imp__accept@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x2e8e): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x2f5f): undefined reference to `_imp__ioctlsocket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_server.c.obj):TCP_server.c:(.text+0x2f89): undefined reference to `_imp__recv@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_client.c.obj):TCP_client.c:(.text+0x21): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_client.c.obj):TCP_client.c:(.text+0xdb): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_client.c.obj):TCP_client.c:(.text+0x26e): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_client.c.obj):TCP_client.c:(.text+0x373): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_client.c.obj):TCP_client.c:(.text+0x5a3): undefined reference to `_imp__send@16'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_client.c.obj):TCP_client.c:(.text+0xc71): more undefined references to `_imp__send@16' follow
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_client.c.obj):TCP_client.c:(.text+0x17b3): undefined reference to `_imp__socket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetcrypto.a(TCP_client.c.obj):TCP_client.c:(.text+0x1965): undefined reference to `_imp__connect@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxdht.a(LAN_discovery.c.obj):LAN_discovery.c:(.text+0x8c8): undefined reference to `GetAdaptersInfo@8'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxdht.a(LAN_discovery.c.obj):LAN_discovery.c:(.text+0x8e8): undefined reference to `GetAdaptersInfo@8'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0xdf): undefined reference to `_imp__WSAStartup@8'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x43c): undefined reference to `_imp__closesocket@4'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x474): undefined reference to `_imp__ioctlsocket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x4d4): undefined reference to `_imp__setsockopt@20'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x52e): undefined reference to `_imp__getsockopt@20'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x56c): undefined reference to `_imp__setsockopt@20'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x6e5): undefined reference to `_imp__sendto@24'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x83b): undefined reference to `_imp__recvfrom@24'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0xa5b): undefined reference to `_imp__WSAStartup@8'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0xafe): undefined reference to `_imp__closesocket@4'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0xf42): undefined reference to `_imp__WSAAddressToStringA@20'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x1036): undefined reference to `_imp__WSAAddressToStringA@20'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x10bd): undefined reference to `_imp__WSAAddressToStringA@20'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x112a): undefined reference to `_imp__WSAAddressToStringA@20'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x1259): undefined reference to `_imp__socket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x1275): undefined reference to `_imp__setsockopt@20'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x132e): undefined reference to `_imp__ioctlsocket@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x145f): undefined reference to `_imp__bind@12'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x14f2): undefined reference to `_imp__WSAStartup@8'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x171b): undefined reference to `_imp__closesocket@4'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x1783): undefined reference to `_imp__getsockopt@20'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x1996): undefined reference to `_imp__WSAAddressToStringA@20'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x1a14): undefined reference to `_imp__WSAStringToAddressA@20'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(network.c.obj):network.c:(.text+0x1b7f): undefined reference to `_imp__WSAStringToAddressA@20'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(util.c.obj):util.c:(.text+0x36d): undefined reference to `pthread_mutexattr_init'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(util.c.obj):util.c:(.text+0x381): undefined reference to `pthread_mutexattr_settype'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(util.c.obj):util.c:(.text+0x3a3): undefined reference to `pthread_mutexattr_destroy'
/home/jenkins/workspace/qTox_build_windows_x86_release/libs/lib/libtoxnetwork.a(util.c.obj):util.c:(.text+0x3c9): undefined reference to `pthread_mutexattr_destroy'
collect2: error: ld returned 1 exit status

@sudden6, it's 482757e

--- a/qtox.pro
+++ b/qtox.pro
@@ -128,8 +128,20 @@ win32 {
     LIBS += -L$$PWD/libs/lib \
             -ltoxav \
             -ltoxcore \
-            -ltoxencryptsave \
-            -lsodium \
+            -ltoxencryptsave
+
+    # must be exactly here, to preserve link order
+    contains(TOX_CMAKE, YES) {
+        LIBS += -ltoxgroup \
+                -ltoxmessenger \
+                -ltoxfriends \
+                -ltoxnetcrypto \
+                -ltoxdht \
+                -ltoxnetwork \
+                -ltoxcrypto
+    }
+
+    LIBS += -lsodium \
             -lvpx \
             -lpthread \
             -lavdevice \
@@ -152,16 +164,6 @@ win32 {
             -lshlwapi \
             -luuid
     LIBS += -lstrmiids # For DirectShow
-
-    contains(TOX_CMAKE, YES) {
-        LIBS += -ltoxgroup \
-                -ltoxmessenger \
-                -ltoxfriends \
-                -ltoxnetcrypto \
-                -ltoxdht \
-                -ltoxnetwork \
-                -ltoxcrypto
-    }
 } else {
     macx {
         BUNDLEID = chat.tox.qtox

